### ISSUE TITLE
[Research] Add PTSD/Trauma Hyperarousal Detection from Frontal EEG (AF3/AF4 Shannon Entropy)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#52]
## Paper

**Title:** Towards predicting PTSD symptom severity using portable EEG-derived biomarkers
**Authors:** Various (Scientific Reports)
**Year:** 2024-2025
**Venue:** Scientific Reports (Nature)
**URL:** https://pmc.ncbi.nlm.nih.gov/articles/PMC11275680/

**Supporting:** PTSD hyperarousal event detection using smartwatch physiological and activity data (PLOS ONE, 2022) — XGBoost with 83% accuracy on smartwatch data

## Key Contribution

First study using portable (14-channel Emotiv EPOC X) EEG to predict PTSD symptom severity (PCL-5 scores). Found significant negative correlations between Shannon entropy at fronto-temporal sites (AF3: r=-0.456, AF4: r=-0.362, T7: r=-0.472) and PTSD severity. Alpha power at AF4 also negatively correlated (r=-0.429). Higher PTSD severity = reduced EEG complexity + lower alpha power. Session duration: only 3 minutes.

Critically: AF3/AF4 are nearly identical positions to Muse 2's AF7/AF8. The biomarkers discovered here are directly measurable on our hardware.

## How It Applies to NeuralDreamWorkshop

NeuralDreamWorkshop already computes spectral entropy and alpha power at AF7/AF8. This research shows these features can serve as trauma/hyperarousal biomarkers with minimal additional work:

1. **Shannon entropy at AF7/AF8** — already computed via `spectral_entropy()` in `eeg_processor.py`
2. **Alpha power at AF8** — already extracted in `extract_band_powers()`
3. **Temporal entropy at TP9/TP10** — T7 correlation (r=-0.472) maps to TP9/TP10

This creates a new "Wellness" feature: a trauma-aware hyperarousal index that flags chronically elevated arousal states, distinct from the current stress_index which measures momentary stress.

### Files to Modify/Create
- `ml/models/hyperarousal_detector.py` — new model (simple threshold-based, like stress detector)
- `ml/api/routes.py` — add `/analyze-hyperarousal` endpoint
- `ml/processing/eeg_processor.py` — add `compute_shannon_entropy()` (time-domain, distinct from spectral entropy)
- `client/src/pages/wellness-dashboard.tsx` — display hyperarousal trend over sessions

### Core Algorithm
```python
def compute_hyperarousal_index(signals: np.ndarray, fs: int = 256) -> dict:
    """
    Compute trauma-related hyperarousal index from 4-channel Muse 2 EEG.
    Based on: AF3/AF4/T7 Shannon entropy negatively correlated with PCL-5.
    
    Lower entropy + lower alpha → higher hyperarousal risk.
    """
    # Channel mapping: ch0=TP9, ch1=AF7, ch2=AF8, ch3=TP10
    
    # 1. Shannon entropy per channel (time-domain, 3-min window recommended)
    from scipy.stats import entropy as shannon_entropy
    af7_entropy = _compute_binned_entropy(signals[1], n_bins=64)
    af8_entropy = _compute_binned_entropy(signals[2], n_bins=64)
    tp9_entropy = _compute_binned_entropy(signals[0], n_bins=64)
    
    # 2. Alpha power at AF8 (r=-0.429 with PTSD severity)
    af8_alpha = extract_band_powers(signals[2], fs)['alpha']
    
    # 3. Hyperarousal index: inverse of normalized entropy + alpha
    # Lower values = more hyperarousal risk
    frontal_entropy = (af7_entropy + af8_entropy) / 2
    temporal_entropy = tp9_entropy
    
    # Normalize against baseline (requires BaselineCalibrator)
    # Raw index: 0 = maximum hyperarousal risk, 1 = calm/complex EEG
    hyperarousal_raw = 1.0 - np.clip(
        0.40 * frontal_entropy_z + 0.30 * temporal_entropy_z + 0.30 * af8_alpha_z,
        0, 1
    )
    
    return {
        "hyperarousal_index": float(hyperarousal_raw),
        "frontal_entropy": float(frontal_entropy),
        "temporal_entropy": float(temporal_entropy),
        "af8_alpha_power": float(af8_alpha),
        "risk_level": "elevated" if hyperarousal_raw > 0.65 else "normal",
        "note": "NOT a clinical diagnostic — for self-awareness only"
    }
```

### Important Ethical Considerations
- This is NOT a PTSD diagnostic tool — must include disclaimers
- Display as "Arousal Regulation" or "Nervous System Balance" — avoid clinical labels
- Recommend professional help if persistent elevation detected
- Never store or share this data without explicit consent

## Expected Performance Impact

- The correlation coefficients (r=-0.43 to -0.47) are moderate but significant
- Expected classification (elevated vs normal hyperarousal): ~70-75% accuracy with 3-min epochs
- Main value: longitudinal trending over sessions, not single-session diagnosis
- Complements existing stress_index by measuring chronic arousal pattern vs momentary stress

## Complexity: LOW

- All required features (entropy, alpha power) already exist in eeg_processor.py
- Shannon entropy computation is trivial (~5 lines of scipy code)
- No new model training needed — threshold-based like stress detector
- Main work: UI page for longitudinal tracking + ethical disclaimers
- Risk: need minimum 3-minute continuous recording for reliable entropy estimate (current sessions may be shorter)

## Dependencies
- None new — scipy already available
- BaselineCalibrator (already implemented) for normalization
